src/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
+	tcollama "github.com/testcontainers/testcontainers-go/modules/ollama"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+)
+
+type TagResponse struct {
+	Models []struct {
+		Name string `json:"name"`
+	} `json:"models"`
+}
+
+type GenerateRequest struct {
+	Model  string `json:"model"`
+	Stream bool   `json:"stream"`
+	Prompt string `json:"prompt"`
+}
+
+func startContainer(image string) (string, error) {
+	ctx := context.Background()
+	ollamaContainer, err := tcollama.Run(ctx, image, testcontainers.CustomizeRequestOption(func(req *testcontainers.GenericContainerRequest) error {
```

Comment:
TIL! This is cool! 😄 
Wished I'd known about tc much earlier. Have written too much code where I had to write the boiler plate for this myself.. :/